File: packages/live-update/ios/Plugin/Classes/Results/ChannelResult.swift
import Foundation
import Capacitor

@objc public class ChannelResult: NSObject, Result {
    private let id: String
    private let name: String

    init(id: String, name: String) {
        self.id = id
        self.name = name
    }

    public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["id"] = id
        result["name"] = name
        return result as AnyObject
    }
}

File: packages/live-update/android/src/main/java/io/capawesome/capacitorjs/plugins/liveupdate/classes/results/ChannelResult.java
package io.capawesome.capacitorjs.plugins.liveupdate.classes.results;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.liveupdate.interfaces.Result;

public class ChannelResult implements Result {

    @NonNull
    private final String id;

    @NonNull
    private final String name;

    public ChannelResult(@NonNull String id, @NonNull String name) {
        this.id = id;
        this.name = name;
    }

    @NonNull
    @Override
    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("id", id);
        result.put("name", name);
        return result;
    }
}
